jlog.c

```diff
@@ -2179,6 +2179,139 @@ int jlog_ctx_read_message(jlog_ctx *ctx, const jlog_id *id, jlog_message *m) {
   }
   return -1;
 }
+int jlog_ctx_bulk_read_messages(jlog_ctx *ctx, const jlog_id *id, const int count, jlog_message *m) {
```

Comment:
Personally, I think the time to optimize this is later. Get it running, then fix it once it's working. I don't want to take any unnecessary risks here by touching an already-in-use code path without extensive testing of that.